lib/chef/provider/cron.rb

```diff
@@ -88,7 +88,11 @@ def load_current_resource
 
       def cron_different?
         CRON_ATTRIBUTES.any? do |cron_var|
-          new_resource.send(cron_var) != current_resource.send(cron_var)
+          if new_resource.send(cron_var).is_a?(Integer)
+            new_resource.send(cron_var).to_s != current_resource.send(cron_var)
+          else
+            new_resource.send(cron_var) != current_resource.send(cron_var)
+          end
```

Comment:
```suggestion
          new_resource.send(cron_var).to_s != current_resource.send(cron_var).to_s
```

Reply:
Problem with this is the `environment` attribute, which can be a hash. `to_s` won't take sorting into account.

Reply:
What if instead we check types - and only coerce them to string if they are not the same?

Reply:
Seems reasonable to me. Just pushed up a revision that will do that instead.